### ISSUE TITLE
Make HttpClient compatible with Browser

### ISSUE THREAD

**Issue by braden-w** [Effect-TS/effect#3322]
### What is the problem this feature would solve?

HTTPClient is currently not working in a browser context.

Currently, the HTTP client works well in a Node.js runtime on the server side, allowing for GET, PUT, POST, and other HTTP methods. However, when attempting to use these methods in a browser environment, the following error occurs:

```
Access to fetch at '...' from origin 'http://localhost:5173' has been blocked by CORS policy: Request header field b3 is not allowed by Access-Control-Allow-Headers in preflight response.
```

Upon examining the code, it appears that additional header fields such as "b3" and "traceparent" are added to the request. I believe these added headers are causing the CORS policy to block the request in the browser (though it could be another issue as well).

### What is the feature you are proposing to solve the problem?

An enhancement for the HTTP client to be fully functional within a browser context. 

Introduce an option to configure the HTTP client to strip out or avoid adding these specific headers ("b3" and "traceparent") when running in a browser context. Maybe we could do one of:

1. Adding a configuration option to enable or disable these headers.
2. Detecting the runtime environment (browser or server) and applying the configuration accordingly.
3. Ensuring that the client-side HTTP requests comply with CORS policies by excluding restricted headers.

I could be completely wrong here and there could be other reasons why my requests are not going through, but I would really like to use Effect's HTTP client on the client side!

### What alternatives have you considered?

Using native browser `fetch` in client side code.

**Comment by fubhy** [comment]
HttpClient works perfectly fine in the browser. The error you are seeing here just means that the server you are communicating with does not allow these headers in its CORS policy.

These headers are essential for distributed tracing and usually desirable. Also in the browser.

If you don't have access to the fix your server CORS configuration (e.g. because it's an external API), you can already disable tracer propagation on your http client using `HttpClient.withTracerPropagation(false)`.

**Comment by braden-w** [comment]
Thank you very much! I was working with a third-party API, so `HttpClient.withTracerPropagation(false)` is really helpful to know!
